language_generators/scala-defref-extractor/src/main/scala/io/bazeltools/buildgen/scaladefref/ScalaSourceEntityExtractor.scala

```diff
@@ -470,8 +513,28 @@ object ScalaSourceEntityExtractor {
     def processScope(s: ScopeState): Writer[DataBlock, Unit] = {
       val defs = s.definedEntities
       val refs = s.nonLocalRefs(getResolve(s))
-
-      Writer.tell(DataBlock("", defs = defs, refs = refs))
+      val stringRefs = refs.map(_.toString)
+      val links = getLinksRecursive(s, Nil).iterator
+        .flatMap { case (src, dests) =>
+          dests.iterator.iterator
```

Comment:
I am pretty sure we don't want `.iterator.iterator`. I think `dests` is a `List` so `List.iterator` is an iterator.

also, I think a for comprehension will actually read better here:

```scala
val linksIter = for {
  sd <- getLinksRecursive(s, Nil).iterator
  (src, dests) = sd
  dest <- dests.iterator
  data = dest.value.split('.').toList
  y <- stringRefs.iterator
  if y.endsWith(data.head)
  tailStr = tail.iterator.map { t => s"${y}.$t" }
  linkTarget <- Iterator.single(e) ++ tailStr
} yield s"link: ${src} -> $linkTarget"

val links = linksIter.to(SortedSet)
```